File: app/src/main/java/com/dpis/module/PackageConfigRepository.java
package com.dpis.module;

import com.dpis.module.templates.TemplateConfigValue;

public final class PackageConfigRepository {
    private final DpisConfigStore store;

    public PackageConfigRepository(DpisConfigStore store) {
        this.store = store;
    }

    public boolean hasRealPackageConfig(String packageName) {
        return store != null && store.hasRealPackageConfig(packageName);
    }

    public boolean writePackageTemplateConfigValue(String packageName, TemplateConfigValue value) {
        return store != null && store.writePackageTemplateConfigValue(packageName, value);
    }
}

File: app/src/main/java/com/dpis/module/config/PackageConfigRepository.kt
package com.dpis.module.config

import com.dpis.module.DpisConfigStore
import com.dpis.module.templates.TemplateConfigValue

class PackageConfigRepository(private val store: DpisConfigStore?) {
    fun hasRealPackageConfig(packageName: String?): Boolean =
        store?.hasRealPackageConfig(packageName) == true

    fun writePackageTemplateConfigValue(
        packageName: String?,
        value: TemplateConfigValue?,
    ): Boolean = store?.writePackageTemplateConfigValue(packageName, value) == true
}
